src/components/Autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,448 @@
+import MultiSelectChips from './MultiSelectChips';
+import { isIosDevice } from './utils/helpers';
+import { AutocompleteProps, OptionType } from './utils/types';
+
+const Autocomplete: React.FC<AutocompleteProps> = ({
+  options,
+  label,
+  placeholder,
+  multiple = false,
+  hideArrow = false,
+  hasClearIcon = false,
+  avoidLocalFiltering = false,
+  noResultsText = 'Non ci sono corrispondenze da mostrare',
+  disabled = false,
+  required = false,
+  loading = false,
+  sx,
+  inputStyle,
+  slots = {},
+  slotProps = {},
+  ariaLabels = {},
+  overridenInputvalue,
+  renderOption,
+  onInputChange,
+  onSelect,
+  setInputValueOnSelect,
+}) => {
+  const [inputValue, setInputValue] = useState<string>(overridenInputvalue || '');
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [activeIndex, setActiveIndex] = useState<number>(-1);
+  const [selectedOptions, setSelectedOptions] = useState<Array<OptionType>>([]);
+
+  const containerRef = useRef<HTMLDivElement>(null);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const listboxRef = useRef<HTMLUListElement>(null);
+  const listboxId = 'autocomplete-listbox';
+  const inputId = 'autocomplete-input';
+
+  const {
+    startIcon: StartIcon,
+    clearIcon: ClearIcon = Close,
+    expandIcon: ExpandIcon = KeyboardArrowDown,
+    collapseIcon: CollapseIcon = KeyboardArrowUp,
+    emptyState: EmptyState,
+    loadingSkeleton: LoadingSkeleton,
+  } = slots;
+
+  const {
+    startIcon: startIconProps = {},
+    clearIcon: clearIconProps = {},
+    expandIcon: expandIconProps = {},
+    collapseIcon: collapseIconProps = {},
+    emptyState: emptyStateProps = {},
+    input: inputProps = {},
+    loadingSkeleton: loadingSkeletonProps = {},
+  } = slotProps;
+
+  const {
+    clearButtonLabel = 'Cancella il testo inserito',
+    collapseButtonLabel = 'Chiudi il menu a tendina',
+    expandButtonLabel = 'Apri il menu a tendina',
+    loadingLabel = 'Caricamento in corso',
+    selectedOptionsLabel = 'Opzioni selezionate',
+  } = ariaLabels;
+
+  const filteredOptions =
+    inputValue.trim() === '' || avoidLocalFiltering
+      ? options
+      : options.filter((option) => option.label.toLowerCase().includes(inputValue.toLowerCase()));
+
+  const handleInputBlur = () => {
+    if (disabled) {
+      return;
+    }
+
+    const focusingAnOption = activeIndex !== -1;
+    const keepMenuOpen = isOpen && isIosDevice();
+    if (!focusingAnOption && !keepMenuOpen) {
+      setIsOpen(false);
+    }
+  };
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (disabled) {
+      return;
+    }
+
+    setInputValue(e.target.value);
+    setIsOpen(true);
+    setActiveIndex(-1);
+    onInputChange?.(e.target.value);
+  };
+
+  const handleOptionSelect = (option: OptionType) => {
+    if (disabled) {
+      return;
+    }
+
+    if (multiple) {
+      const isAlreadySelected = selectedOptions.some((selected) => selected.id === option.id);
```

Comment:
If we remove the constraint on the Option model, we will be not sure that the id prop exists. So we can export an optional method like isOptionEqualToValue (as in mui). If it is not defined we can do the === comparison. Emit an error if more than on option match the comparison